tests/test_dataset.py

```diff
@@ -146,17 +147,38 @@ def test_mdp_dataset(
         assert episode is dataset.episodes[i]
 
     # check append
+    new_size = 2
     dataset.append(observations, actions, rewards, terminals)
-    assert len(dataset) == 2 * n_episodes
-    assert dataset.observations.shape == (2 * data_size, observation_size)
-    assert dataset.rewards.shape == (2 * data_size,)
-    assert dataset.terminals.shape == (2 * data_size,)
+    assert len(dataset) == new_size * n_episodes
+    assert dataset.observations.shape == (
+        new_size * data_size,
+        observation_size,
+    )
+    assert dataset.rewards.shape == (new_size * data_size,)
+    assert dataset.terminals.shape == (new_size * data_size,)
     if discrete_action:
-        assert dataset.actions.shape == (2 * data_size,)
+        assert dataset.actions.shape == (new_size * data_size,)
     else:
-        assert dataset.actions.shape == (2 * data_size, action_size)
+        assert dataset.actions.shape == (new_size * data_size, action_size)
+
+    # check append if discrete action and number of actions grow
+    if discrete_action:
+        new_size += 1
```

Comment:
Thanks for the test. But, I actually want you to test specifically if `dataset.get_action_size()` is updated based on the new data. You can simplify this part.

Reply:
Ok, I added the specific check for dataset.get_action_size() == old size + new actions